types/socketcluster-server/scserversocket.d.ts

```diff
@@ -29,13 +29,33 @@ declare class SCServerSocket extends Emitter {
 
     constructor(id: string, server: SCServer, socket: WebSocket);
 
+    // tslint:disable-next-line ban-types
+    on(event: string, listener: Function): this;
```

Comment:
As discussed [here](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes), we'd strongly prefer not using the type `Function`; even (...args: any[]) => any) is preferable.

Reply:
This is a copy of the definition from Emitter, which I added so the general case works (see the test). I could change it here, but even better would be if you can tell me how to write the type definition in a way that the "base" method definition works as well.

Reply:
The best way to think about it is that every time you see `Function`, it should really be something like `(...args: any[]) => any)`. While the former has all the same properties as the latter (and therefore has the same shape and is compatible from a type comparison perspective), it's ultimately an object type, and therefore makes type checking a little harder because it lacks things like a notion of a return type or arguments. So if you use the latter form here, it will certainly be compatible with the definition from `Emitter` and will allow any function a user ends up passing there. Would also be great to fix the definition of `Emitter`, but I don't want to cause too much scope creep for your PR.

Reply:
Note that if this feels a bit verbose, you can always introduce something like 
`type AnyFunction = (...args: any[]) => any` and use that.